ext/spl/spl_directory.stub.php

```diff
@@ -190,101 +190,101 @@ class SplFileObject extends SplFileInfo implements RecursiveIterator, SeekableIt
 
     /**
      * @param int $wouldBlock
-     * @return bool
+     * @tentative-return-type
      */
-    public function flock(int $operation, &$wouldBlock = null) {}
+    public function flock(int $operation, &$wouldBlock = null): bool {}
 
-    /** @return bool */
-    public function fflush() {}
+    /** @tentative-return-type */
+    public function fflush(): bool {}
 
-    /** @return int|false */
-    public function ftell() {}
+    /** @tentative-return-type */
+    public function ftell(): int|false {}
 
-    /** @return int */
-    public function fseek(int $offset, int $whence = SEEK_SET) {}
+    /** @tentative-return-type */
+    public function fseek(int $offset, int $whence = SEEK_SET): int {}
 
-    /** @return string|false */
-    public function fgetc() {}
+    /** @tentative-return-type */
+    public function fgetc(): string|false {}
 
-    /** @return int */
-    public function fpassthru() {}
+    /** @tentative-return-type */
+    public function fpassthru(): int {}
 
-    /** @return array|int|false|null */
-    public function fscanf(string $format, mixed &...$vars) {}
+    /** @tentative-return-type */
+    public function fscanf(string $format, mixed &...$vars): array|int|null {}
 
-    /** @return int|false */
-    public function fwrite(string $data, int $length = 0) {}
+    /** @tentative-return-type */
+    public function fwrite(string $data, int $length = 0): int|false {}
 
-    /** @return array */
-    public function fstat() {}
+    /** @tentative-return-type */
+    public function fstat(): array {}
 
-    /** @return bool */
-    public function ftruncate(int $size) {}
+    /** @tentative-return-type */
+    public function ftruncate(int $size): bool {}
 
-    /** @return string|array|false */
-    public function current() {}
+    /** @tentative-return-type */
+    public function current(): string|array|false {}
 
-    /** @return int */
-    public function key() {}
+    /** @tentative-return-type */
+    public function key(): int {}
 
-    /** @return void */
-    public function next() {}
+    /** @tentative-return-type */
+    public function next(): void {}
 
-    /** @return void */
-    public function setFlags(int $flags) {}
+    /** @tentative-return-type */
+    public function setFlags(int $flags): void {}
 
-    /** @return int */
-    public function getFlags() {}
+    /** @tentative-return-type */
+    public function getFlags(): int {}
 
-    /** @return void */
-    public function setMaxLineLen(int $maxLength) {}
+    /** @tentative-return-type */
+    public function setMaxLineLen(int $maxLength): void {}
 
-    /** @return int */
-    public function getMaxLineLen() {}
+    /** @tentative-return-type */
+    public function getMaxLineLen(): int {}
 
-    /** @return bool */
-    public function hasChildren() {}
+    /** @tentative-return-type */
+    public function hasChildren(): bool {}
 
-    /** @return null */
+    /** @return mixed */
     public function getChildren() {}
```

Comment:
This is a bit weird implementation. So I'm not sure exactly what the return type should be.

I now see that `spl_recursive_it_move_forward_ex` invokes this method, and the error message indicates that `RecursiveIterator` should be returned.. Which is something even the parent method doesn't do.

Reply:
As mentioned on the other review, I believe this should use `?RecursiveIterator`.